src/deps/PDFWriter/PDFPageInput.cpp

```diff
@@ -181,7 +181,13 @@ void PDFPageInput::SetPDFRectangleFromPDFArray(PDFArray* inPDFArray,PDFRectangle
 	RefCountPtr<PDFObject> lowerLeftY(inPDFArray->QueryObject(1));
 	RefCountPtr<PDFObject> upperRightX(inPDFArray->QueryObject(2));
 	RefCountPtr<PDFObject> upperRightY(inPDFArray->QueryObject(3));
-	
+	if (!lowerLeftX || !lowerLeftY || !upperRightX || !upperRightY)
+	{
+		// not sure if just a return is a good idea here.
```

Comment:
not sure if this is a good idea

Reply:
well its all in teh unexpected area so any decision is good.
Inspecting all usages there's a pattern of checking for an array, then if it's 4 and then either going for a default or calling SetPDFRectangleFromArray.

so that array check+ 4 check can be moved into SetPDFRectangleFromArray for better reuse and that it'd return status. then it's `if(SetPDFRectangleFromArray(....) == eFailure) do_the_default.